src/Compilers/CSharp/Test/Syntax/Parsing/CollectionExpressionParsingTests.cs

```diff
@@ -18758,4 +18758,1991 @@ public void CollectionExpression_KeywordKey_Illegal5()
+                    N(SyntaxKind.AttributeArgumentList);
+                    {
+                        N(SyntaxKind.OpenParenToken);
+                        N(SyntaxKind.CloseParenToken);
+                    }
+                }
+                N(SyntaxKind.CloseBracketToken);
+            }
+            N(SyntaxKind.AsyncKeyword);
+            N(SyntaxKind.Parameter);
+            {
+                N(SyntaxKind.IdentifierToken, "a");
+            }
+            N(SyntaxKind.EqualsGreaterThanToken);
+            N(SyntaxKind.IdentifierName);
+            {
+                N(SyntaxKind.IdentifierToken, "b");
+            }
+        }
+        EOF();
+    }
+
+    [Fact]
+    public void NotWithElement18()
+    {
+        UsingExpression("[with()] (a) => b");
+
+        N(SyntaxKind.ParenthesizedLambdaExpression);
+        {
+            N(SyntaxKind.AttributeList);
+            {
+                N(SyntaxKind.OpenBracketToken);
+                N(SyntaxKind.Attribute);
+                {
+                    N(SyntaxKind.IdentifierName);
+                    {
+                        N(SyntaxKind.IdentifierToken, "with");
+                    }
+                    N(SyntaxKind.AttributeArgumentList);
+                    {
+                        N(SyntaxKind.OpenParenToken);
+                        N(SyntaxKind.CloseParenToken);
+                    }
+                }
+                N(SyntaxKind.CloseBracketToken);
+            }
+            N(SyntaxKind.ParameterList);
+            {
+                N(SyntaxKind.OpenParenToken);
+                N(SyntaxKind.Parameter);
+                {
+                    N(SyntaxKind.IdentifierToken, "a");
+                }
+                N(SyntaxKind.CloseParenToken);
+            }
+            N(SyntaxKind.EqualsGreaterThanToken);
+            N(SyntaxKind.IdentifierName);
+            {
+                N(SyntaxKind.IdentifierToken, "b");
+            }
+        }
+        EOF();
+    }
+
+    [Fact]
+    public void WithElement1()
+    {
+        UsingExpression("[with(]",
+            // (1,7): error CS1026: ) expected
+            // [with(]
+            Diagnostic(ErrorCode.ERR_CloseParenExpected, "]").WithLocation(1, 7),
+            // (1,8): error CS1003: Syntax error, ']' expected
+            // [with(]
+            Diagnostic(ErrorCode.ERR_SyntaxError, "").WithArguments("]").WithLocation(1, 8));
+
+        N(SyntaxKind.CollectionExpression);
+        {
+            N(SyntaxKind.OpenBracketToken);
+            N(SyntaxKind.WithElement);
+            {
+                N(SyntaxKind.WithKeyword);
+                N(SyntaxKind.ArgumentList);
+                {
+                    N(SyntaxKind.OpenParenToken);
+                    M(SyntaxKind.CloseParenToken);
+                }
+            }
+            M(SyntaxKind.CloseBracketToken);
+        }
+        EOF();
+    }
+
+    [Fact]
+    public void WithElement2()
+    {
+        UsingExpression("[with()]");
```

Comment:
This looks like a breaking change. Let's document it.
Should it be made LangVer-dependent?

Reply:
We can doc, but def want to run by ldm first.  Our investigations here indicate no usage of this in the wild.  So we're going to see if we can get buyoff on this just being a unilateral break .

Reply:
this is part of hte final syntax/sign-off discussion there.

Reply:
> Should it be made LangVer-dependent?

This is currently an [open question](https://github.com/dotnet/csharplang/blob/main/proposals/collection-expression-arguments.md#arguments-with-earlier-language-version).

Reply:
Since the PR implements a break, let's add a PROTOTYPE marker (to test with LangVer and document break).

Reply:
open LDM question.

Reply:
> Since the PR implements a break, let's add a PROTOTYPE marker (to test with LangVer and document break).

Done.

Reply:
```suggestion
        // PROTOTYPE test with LangVer and document break
        UsingExpression("[with()]");
```
My reasoning for pushing on this: If LDM were to approve the break as it is implemented by this PR, so nobody revisits this code, who is going to remember that we didn't document it? Usually, PRs that implement breaks also document them.